meta-mender-core/classes/mender-setup.bbclass

```diff
@@ -211,11 +211,16 @@ IMAGE_CLASSES += "mender-part-images mender-ubimg mender-artifactimg mender-data
 IMAGE_NAME = "${IMAGE_BASENAME}-${MENDER_DEVICE_TYPE}${IMAGE_VERSION_SUFFIX}"
 IMAGE_LINK_NAME = "${IMAGE_BASENAME}-${MENDER_DEVICE_TYPE}"
 
-# MENDER_FEATURES_ENABLE and MENDER_FEATURES_DISABLE map to
-# DISTRO_FEATURES_BACKFILL and DISTRO_FEATURES_BACKFILL_CONSIDERED,
-# respectively.
-DISTRO_FEATURES_BACKFILL_append = " ${MENDER_FEATURES_ENABLE}"
-DISTRO_FEATURES_BACKFILL_CONSIDERED_append = " ${MENDER_FEATURES_DISABLE}"
+def mender_feature_is_enabled(feature, if_true, if_false, d):
+    in_enable = bb.utils.contains('MENDER_FEATURES_ENABLE', feature, True, False, d)
+    in_disable = bb.utils.contains('MENDER_FEATURES_DISABLE', feature, True, False, d)
+
+    if in_enable and not in_disable:
+        return if_true
+    else:
+        return if_false
+
+inherit ${@mender_feature_is_enabled('mender-client-install', 'mender-setup-distro-features', '', d)}
```

Comment:
I don't think you need this complicated workaround. Check out my proposed fix in https://github.com/drewmoseley/meta-mender/pull/2.

Reply:
Testing that now.